### PR TITLE
WIP feature/paging

### PR DESCRIPTION
**Pull request by alonzopt**
Still a little messy/light on comments but passes read-all tests for users, patients, and calendar events.

Had to move base `PageIterator` class above `Collection` in `models/base.py` to make vscode respect the reference.

`PageableIterator` and `MetaInfoIterator` are for existing the paging types, `MetaIterator` is for chat. Class names are from their key in the response, very willing to change them.

Will need to rebase for CDT stuff, but at a glance the 4th paging style should be pretty easy to make a Iterator class for.

### CODE REVIEW

**Comment by alonzopt** [comment]
> What's the purpose behind the getter/setter methods here? Also, if we do need to use getters/setter methods, I'd suggest using this pattern:
> 
> ```python
> @property
> def foo(self):
>     return self._foo
> 
> @foo.setter
> def foo(self, value):
>     self._foo = value
> ```

`_set_last` can be changed to a setter since that's going to be universal (at least for the 4 types we have now).`_set_page` is a little more complicated since the token based pagination (`MetaIterator`) doesn't use pages so that needed to be broken out. It might just be easier to merge the functions into something named like `_update_self_after_request(meta)` and the subclasses can do whatever they need in that.

The get methods can probably just be moved to variables, I had started with the sets and was probably overthinking the design and trying to keep that stuff hidden when it doesn't need to be.

**Comment by alonzopt** [comment]
Code coverage dropping is due to the inclusion of the pagination class for chat. Did not want to comment out a class for something that will be in use soon.

File: welkin/models/pagination.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+from welkin.models.base import PageIterator
+
+
+class PageableIterator(PageIterator):
+    """
+    Most common paging class
+    """
+
+    def __init__(self, collection, resource, method, size=20, *args, **kwargs):
```
Comment (samamorgan) [inline welkin/models/pagination.py:9]: No need to have all of the arguments listed in these subclasses. You can just do:

```python
def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
    ...
```

File: welkin/models/pagination.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    Paging class for token based paging
+    """
+
+    def __init__(self, collection, resource, method, size=20, *args, **kwargs):
+        super().__init__(collection, resource, method, size, *args, **kwargs)
+        self.meta_key = "meta"
+        self.meta_dict = {
+            "nextPageToken": None,
+            "prevPageToken": None,
+            "pageSize": 20,
```
Comment (samamorgan) [inline welkin/models/pagination.py:74]: Should this be `"pageSize": self.size`?

**Review by samamorgan — CHANGES_REQUESTED** [review]
Great work! Have a few changes commented. Feel free to push back.

**Comment by alonzopt** [comment]
All suggestions were good and have been implemented.

Reminder the test coverage drops because chat pagination class is included in the pagination file.
Test coverage passes in chat PR https://github.com/Lightmatter/welkin-health/pull/26

**Review by samamorgan — APPROVED** [review]
LGTM!

**Pull request merged**